hw/ip/i2c/doc/_index.md

```diff
@@ -142,6 +148,34 @@ Typically every byte transmitted must also receive an ACK signal, and the IP wil
 However, there are some I2C commands which do not require an ACK.
 In those cases this flag should be asserted with FBYTE indicating no ACK is expected and no interrupt should be raised if the ACK is not received. 
 
+### Target Address Registers
+
+I2C target device is assigned two 7-bit address and 7-bit mask pairs.
+The target device accepts a transaction if the result of the bitwise AND operation performed on the transaction address sent by the host and a mask matches the assigned address corresponding to the mask.
+Thus, with the masks set to all ones (0x7F), the target device will respond to either of the two assigned unique addresses and no other.
+If the mask and the assigned address both have zeros in a particular bit position, that bit will be a match regardless of the value of that bit received from the host.
+Note that if, in any bit position, the mask has zero and the assigned address has one, no transaction can match and such mask/address pair is effectively diasabled.
+The assigned address and mask pairs are set in registers {{< regref TARGET_ID0.ADDRESS >}}, {{< regref TARGET_ID0.MASK >}}, {{< regref TARGET_ID1.ADDRESS >}}, and {{< regref TARGET_ID1.MASK >}}. 
+
+### Acquired Formatted Data
+
+This section applies to I2C in the target mode.
+When the target accepts a transaction, it inserts the transaction address, read/write bit, and START signal sent by the host into ACQ FIFO where they can be accessed by software.
+ACQ FIFO output corresponds to {{< regref ACQDATA >}}.
+If the transaction is a write operation (R/W bit = 0), the target proceeds to read bytes from the bus and insert them into ACQ FIFO until the host terminates the transaction by sending a STOP or RESTART signal.
+A STOP indicator is inserted into ACQ FIFO as the next entry following the last byte received, in which case other bits are junk.
+A RESTART is indicated by inserting both STOP and START indicators into ACQ FIFO as the next entry; the remaining bits are junk.
+If the transaction is a read operation (R/W bit = 1), the target pulls bytes out of TX FIFO and transmits them to the bus until the host signals the end of the transfer by sending a NACK signal.
+If TX FIFO holds no data, the target will hold SCL low to stretch the clock and give software time to write data bytes into TX FIFO.
+TX FIFO input corresponds to {{< regref TXDATA >}}.
+Typically, a NACK signal is followed by a STOP or RESTART signal and the IP will raise an exception if the host sends a STOP signal after an ACK.
+An ACK/NACK signal is inserted into the ACQ FIFO as the first bit (bit 0), in the same entry with a STOP or RESTART signal. 
```

Comment:
So I'm confused here.  Does this mean that after every read, the ACQ FIFO will be filled with n-ACK's, and a NACK?  Is it then firmware's job to pull these out and dump them? Are these useful to Firmware in any way?

Reply:
While the host is reading, ACK/NACK bits will not be written into ACQ FIFO. Once the host sends a STOP signal, ACK/NACK will be written into the next entry, together with the STOP.

Reply:
So only when the host is writing then?

Reply:
ACK/NACK will be inserted into ACQ FIFO only when the host is reading; the host is sending ACK/NACK.